### PR TITLE
added waitForActive and waitForOutput to download options on the cli

### PR DESCRIPTION
**Pull request by pujal0909**
**What It Does**

Adds the waitForActive & waitForOutput options for download zosjobs to the CLI.

**How to Test**

To test for active status, run zowe zos-jobs download output --wfa. To test for output status, run zowe zos-jobs download output --wfo.

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

The SDK portion of this issue is a separate PR (#2326). This PR is intended to be merged into download-jobs-enhancement before download-jobs-enhancement is merged to master.

### CODE REVIEW

File: packages/cli/src/zosjobs/download/download-output/Output.definition.ts [hunk 1]
```diff
@@ -73,7 +73,18 @@ export const OutputDefinition: ICommandDefinition = {
                 "data conversion is performed using the file encoding specified.",
             type: "string",
             conflictsWith: ["binary", "record"]
-        }
+        },
+        {
+            name: "wait-for-active", aliases: ["wfa"],
+            description: "Wait for the job to enter ACTIVE status before completing the command.",
+            type: "boolean",
+            conflictsWith: ["wait-for-output", "directory"]
```
Comment (traeok) [inline packages/cli/src/zosjobs/download/download-output/Output.definition.ts:81]: I think `wait-for-output` should stay in the `conflictsWith` array, but should `directory` be removed? If a user is downloading output for a job, they might want to specify a directory - I don't believe the `wait-for` parameters will have an adverse effect when specified in combination with `directory`.

**Review by traeok — COMMENTED** [review]
Looks good to me, thanks Pujal! I just had a question regarding the `conflictsWith` list for the `wait-for-active` parameter.

**Review by zFernand0 — DISMISSED** [review]
changes LGTM! 😋 

left a few suggestions 😅

File: packages/cli/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to the Zowe CLI package will be documented in this file.
 
+## Recent Changes
+
+- Enhancement: Added --wait-For-Active and --wait-For-Output to download options on zosjobs. [#2328](https://github.com/zowe/zowe-cli/pull/2328)
```
Comment (zFernand0) [inline packages/cli/CHANGELOG.md:7]:
```suggestion
- Enhancement: Added `--wait-for-active` and `--wait-for-output` to download options on zosjobs. [#2328](https://github.com/zowe/zowe-cli/pull/2328)
```

**Review by zFernand0 — COMMENTED** [review]
nitpicky comment (about lowercasing the option names)

changes LGTM though 😅

**Comment by traeok** [comment]
I noticed tests are failing, but since `main` isn't the base branch, this could be resolved in the other PR 🙂 aside from that, LGTM

**Review by zFernand0 — APPROVED** [review]

**Pull request merged**